modules/service/src/main/scala/lucuma/odb/service/UserInvitationService.scala

```diff
@@ -55,14 +55,23 @@ object UserInvitationService:
   def instantiate[F[_]: MonadCancelThrow](emailConfig: Config.Email, httpClient: Client[F])(using Services[F]): UserInvitationService[F] =
     new UserInvitationService[F]:
 
-      def sendInvitation(input: CreateUserInvitationInput, invitation: UserInvitation, pid: Program.Id)(
-        using Transaction[F]): F[Result[EmailId]] = {
+      def sendInvitation(
+        input:      CreateUserInvitationInput,
+        invitation: UserInvitation,
+        pid:        Program.Id,
+        targetRole: ProgramUserRole
+      )(using Transaction[F]): F[Result[EmailId]] = {
+
+        val (action, preposition) = targetRole match
+          case ProgramUserRole.External => ("share data", "from")
+          case _                        => ("collaborate", "on")
+
         val subject: NonEmptyString = NonEmptyString.unsafeFrom(
-          s"Invitation to collaborate from ${user.displayName}"
+          s"Invitation to $action from ${user.displayName}"
         )
 
         val textMessage: NonEmptyString = NonEmptyString.unsafeFrom(
-          s"""${user.displayName} has invited you to collaborate on a Gemini proposal. To accept this invitation go to ${emailConfig.exploreUrl} to log in, then from the upper right menu select "Redeem Invitations" and enter the following key:
+          s"""${user.displayName} has invited you to $action $preposition a Gemini proposal. To accept this invitation go to ${emailConfig.exploreUrl} to log in, then from the upper right menu select "Redeem Invitations" and enter the following key:
```

Comment:
What happens in Explore after a user redeems an invitation?  In the case of an external user, they won't be able to see the program so they cannot be redirected to it.

Reply:
> What happens in Explore after a user redeems an invitation? In the case of an external user, they won't be able to see the program so they cannot be redirected to it.

I'd have to check and make sure, but I'm pretty sure that they get directed to that program. If they don't actually have access to that program the program they would get the "Manage Programs" dialog instead.